azurerm/data_source_firewall.go

```diff
@@ -96,8 +95,8 @@ func dataSourceArmFirewallRead(d *schema.ResourceData, meta interface{}) error {
 
 	if props := read.AzureFirewallPropertiesFormat; props != nil {
 		ipConfigs := flattenArmFirewallIPConfigurations(props.IPConfigurations)
```

Comment:
```suggestion
	if err := d.Set("ip_configuration", flattenArmFirewallIPConfigurations(props.IPConfigurations)); err != nil { 
		return fmt.Errorf("Error setting `ip_configurations`: %+v", err) 
	}
```

Reply:
yes, but it's their original coding style. Anyway, it has been modified by your suggestion both in resource and datasource.